package no.nav.oebs.pdlhendelser.db.repository;

import java.util.List;

import org.springframework.stereotype.Repository;

import no.nav.oebs.pdlhendelser.db.entity.Livshendelse;

/**
 * Grensesnitt for repository som håndterer dataaksess mot livshendelse-tabellen. Metodene implementeres automatisk av Spring
 * Data.
 */
@Repository
public interface LivshendelseRepository extends HendelseRepository<Livshendelse, Long> {

	/**
	 * Finner livshendelser med spesifisert hendelse ID og opplysningstype, unntatt hendelser med spesifisert status.
	 *
	 * @return En liste med alle livshendelser som ble funnet; ellers en tom liste.
	 */
	List<Livshendelse> findByHendelseIdAndHendelseOpplysningstypeAndStatusNotIn(String hendelseId, String opplysningstype,
			List<String> statuser);

	/**
	 * Finner livshendelser med spesifisert personidenter og opplysningstype, samt at hendelsene må være nyere enn spesifisert
	 * ID-verdi.
	 *
	 * @return En liste av alle livshendelser som oppfyller kriteriene; ellers en tom liste.
	 */
	List<Livshendelse> findByHendelsePersonidenterAndHendelseOpplysningstypeAndIdGreaterThan(String hendelsePersonidenter,
			String hendelseOpplysningstype, Long id);
}
